argocd/datadog_checks/argocd/check.py

```diff
@@ -0,0 +1,105 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from six import PY2
+
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheckV2
+from datadog_checks.base.constants import ServiceCheck
+
+from .config_models import ConfigMixin
+from .metrics import API_SERVER_METRICS, APPLICATION_CONTROLLER_METRICS, REPO_SERVER_METRICS
+
+API_SERVER_NAMESPACE, APP_CONTROLLER_NAMESPACE, REPO_SERVER_NAMESPACE = [
+    'argocd.api_server',
+    'argocd.app_controller',
+    'argocd.repo_server',
+]
+
+
+class ArgocdCheck(OpenMetricsBaseCheckV2, ConfigMixin):
+    DEFAULT_METRIC_LIMIT = 0
+
+    def __init__(self, name, init_config, instances):
+        if PY2:
+            raise ConfigurationError(
+                "This version of the integration is only available when using py3. "
+                "Check https://docs.datadoghq.com/agent/guide/agent-v6-python-3 "
+                "for more information."
+            )
+        super(ArgocdCheck, self).__init__(name, init_config, instances)
+        self.check_initializations.appendleft(self._parse_config)
+        self.check_initializations.append(self.configure_additional_transformers)
+
+    def _parse_config(self):
+        self.scraper_configs = []
+        app_controller_endpoint = self.instance.get("app_controller_endpoint")
+        api_server_endpoint = self.instance.get("api_server_endpoint")
+        repo_server_endpoint = self.instance.get("repo_server_endpoint")
+        if not app_controller_endpoint and not repo_server_endpoint and not api_server_endpoint:
+            raise ConfigurationError(
+                "Must specify at least one of the following:"
+                "`app_controller_endpoint`, `repo_server_endpoint` or `api_server_endpoint`."
+            )
+
+        if app_controller_endpoint:
+            self.scraper_configs.append(
```

Comment:
This is slightly personal preference, but I would suggest keeping only the parts that differ inside the conditional:

```python
if api_server_endpoint:
    config_spec = (app_controller_endpoint, APP_CONTROLLER_NAMESPACE, APPLICATION_CONTROLLER_METRICS)
... # the rest of the if-clauses
self.scraper_configs.append(
    self._generate_config(*config_spec)
)
```

This may also let you get rid of `_generate_config` and incorporate its logic into this method.